### ISSUE TITLE
Bug: default ecmaVersion is set to 2025

### ISSUE THREAD

**Issue by vladshcherbin** [eslint/eslint#19143]
### Environment

Node version: 22.10.0
pnpm version: 9.12.3
ESLint version: 9.15.0
Operating System: macOS 15.0.1

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
<!-- Paste your configuration here -->
```
</details>

```js
import { loadESLint } from 'eslint'

const ESLint = await loadESLint()
const eslint = new ESLint()
const config = await eslint.calculateConfigForFile('*.js')

console.log(config.languageOptions.ecmaVersion)

// 2025
```

### What did you expect to happen?

> ecmaVersion (default: "latest") - Indicates the ECMAScript version of the code being linted, determining both the syntax and the available global variables. Set to 3 or 5 for ECMAScript 3 and 5, respectively. Otherwise, you can use any year between 2015 to present. In most cases, we recommend using the default of "latest" to ensure you’re always using the most recent ECMAScript version.

According to docs it should be `latest` and it was `latest` before 9.15.0

### What actually happened?

`ecmaVersion` changed from `latest` to `2025`

### Link to Minimal Reproducible Example

https://github.com/vladshcherbin/eslint-9-15

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

![Image](https://github.com/user-attachments/assets/73ece203-2fd3-4541-9c86-d5d4179550ed)

**Comment by mdjermanovic** [comment]
This works as intended. `ESLint#calculateConfigForFile()` returns the final normalized config for the file.

`ecmaVersion: "latest"` (default) is normalized to the latest version supported by ESLint, which is currently `2025`.

**Comment by vladshcherbin** [comment]
Good to know, before 9.15 `calculateConfigForFile` returned `latest` instead of 2025. Thanks 😉

**Comment by mdjermanovic** [comment]
Yes, in < v9.15.0 some normalizations were performed for each file right before linting. In v9.15.0, those normalizations are done directly in the calculated config object (which is typically shared between multiple files) for performance reasons.
